js/msp/msp_helperfunctions.js

```diff
@@ -0,0 +1,1813 @@
+'use strict';
+MSP.listen(MSP_process_data);
+function MSP_process_data (dataHandler) {
```

Comment:
I agree on the isolation of 'protocol' and 'translation'. My comment was more about making all the 'translation' functions into global functions, instead of bundling them into one object, or better prototype.
Totally agreed re MSP.crunch() and process_data, and that we should work towards replacing them with something better.